src/org/labkey/test/components/glassLibrary/grids/DetailDataPanel.java

```diff
@@ -0,0 +1,122 @@
+package org.labkey.test.components.glassLibrary.grids;
+
+import org.labkey.test.Locator;
+import org.labkey.test.components.Component;
+import org.labkey.test.components.WebDriverComponent;
+import org.openqa.selenium.WebDriver;
+import org.openqa.selenium.WebElement;
+
+import java.util.Optional;
+
+
+/**
+ * This wraps DetailPanelHeader.tsx
+ */
+public class DetailDataPanel extends WebDriverComponent<DetailDataPanel.ElementCache>
+{
+    private final WebElement _el;
+    private final WebDriver _driver;
+    private String _title;
+
+    protected DetailDataPanel(WebElement element, WebDriver driver)
+    {
+        _el = element;
+        _driver = driver;
+    }
+
+    @Override
+    public WebElement getComponentElement()
+    {
+        return _el;
+    }
+
+    @Override
+    public WebDriver getDriver()
+    {
+        return _driver;
+    }
+
+    public String getTitle()
+    {
+        if (_title == null)
+            _title = elementCache().heading.getText();
+        return _title;
+    }
+
+    /**
+     * When the logged-in user does not have edit permissions, the edit button will not be present
+     * @return
+     */
+    public boolean isEditable()
+    {
+        return elementCache().editButton().isPresent();
+    }
+
+    public DetailTableEdit clickEdit()
+    {
+        String title = getTitle();
+        if (!isEditable())
+            throw new IllegalStateException("The current pane (with title ["+title+"]) is not editable");
+
+        elementCache().editButton().get().click();
+
+        return new DetailTableEdit.DetailTableEditFinder(getDriver())
+                .withTitle("Editing " + title).waitFor();
+    }
+
+    public DetailTable table()
+    {
+        return  elementCache().detailTable;
+    }
+
+    @Override
+    protected ElementCache newElementCache()
+    {
+        return new ElementCache();
+    }
+
+    protected class ElementCache extends Component<?>.ElementCache
+    {
+        final WebElement heading = Locator.tagWithClass("div", "detail__edit--heading").findWhenNeeded(this);
+        public Optional<WebElement> editButton()
```

Comment:
Really?  I thought that referencing it in order to call 'isDisplayed()' would trigger initialization... which (if it's not present) means having to handle the NoSuchElementException.
But yeah, having it be optional here is to support the case(s) where the user doesn't have edit permissions

Reply:
Finding a lazy element (findWhenNeeded) handles the NoSuchElementException if you call isDisplayed, it will return false in that case.
Either way it is not a big deal, optional works just as well.